tilequeue/rawr.py

```diff
@@ -0,0 +1,254 @@
+                            resp['ResponseMetadata']['HTTPStatusCode'])
+        failed_messages = resp.get('Failed')
+        if failed_messages:
+            # TODO maybe retry failed messages if not sender's fault? up to a
+            # certain maximum number of attempts?
+            # http://boto3.readthedocs.io/en/latest/reference/services/sqs.html#SQS.Client.send_message_batch # noqa
+            raise Exception('Messages failed to send to sqs: %s' %
+                            len(failed_messages))
+
+    def read(self):
+        """read a single message from the queue"""
+        resp = self.sqs_client.receive_message(
+            QueueUrl=self.queue_url,
+            MaxNumberOfMessages=1,
+            AttributeNames=('SentTimestamp',),
+        )
+        if resp['ResponseMetadata']['HTTPStatusCode'] != 200:
+            raise Exception('Invalid status code from sqs: %s' %
+                            resp['ResponseMetadata']['HTTPStatusCode'])
+        msgs = resp['Messages']
+        assert len(msgs) == 1
+        msg = msgs[0]
+        payload = msg['Body']
+        handle = msg['ReceiptHandle']
+        timestamp = msg['Attributes']['SentTimestamp']
+        metadata = dict(timestamp=timestamp)
+        msg_handle = MessageHandle(handle, payload, metadata)
+        return msg_handle
+
+    def done(self, msg_handle):
+        """acknowledge completion of message"""
+        self.sqs_client.delete_message(
+            QueueUrl=self.queue_url,
+            ReceiptHandle=msg_handle.handle,
+        )
+
+
+def process_expiry(rawr_queue, msg_marshaller, group_by_zoom, coords):
+    """enqueue coords from expiry grouped by parent zoom"""
+    grouped_by_zoom = defaultdict(list)
+    for coord in coords:
+        assert group_by_zoom <= coord.zoom
+        parent = coord.zoomTo(group_by_zoom).container()
+        parent_coord_int = coord_marshall_int(parent)
+        grouped_by_zoom[parent_coord_int].append(coord)
+
+    payloads = (msg_marshaller.marshall(coords)
+                for _, coords in grouped_by_zoom.iteritems())
+    rawr_queue_batch_size = 10
+    for payloads_chunk in grouper(payloads, rawr_queue_batch_size):
+        rawr_queue.send(payloads_chunk)
+
+
+def common_parent(coords, parent_zoom):
+    """
+    Return the common parent for coords
+
+    Also check that all coords do indeed share the same parent coordinate.
+    """
+    parent = None
+    for coord in coords:
+        assert parent_zoom <= coord.zoom
+        coord_parent = coord.zoomTo(parent_zoom).container()
+        if parent is None:
+            parent = coord_parent
+        else:
+            assert parent == coord_parent
+    assert parent is not None, 'No coords?'
+    return parent
+
+
+def convert_coord_object(coord):
+    """Convert ModestMaps.Core.Coordinate -> raw_tiles.tile.Tile"""
+    assert isinstance(coord, Coordinate)
+    return Tile(coord.zoom, coord.column, coord.row)
```

Comment:
`Tile` was intended to take `int` coordinates. Would it be worth either asserting that here, and/or calling `.container()` on the `Coordinate`?

Reply:
Good call, updated in 701ab96.